nexus/reconfigurator/planning/src/blueprint_builder.rs

```diff
@@ -353,30 +366,26 @@ impl<'a> BlueprintBuilder<'a> {
         // of Nexus instances), but wouldn't be ideal if we have many resources
         // we need to skip. We could do something smarter here based on the sets
         // of used resources we built above if needed.
-        let nexus_v4_ips = Box::new(
+        let nexus_v4_ips = AvailableIterator::new(
```

Comment:
I ended up not using the extra functionality provided by `AvailableIterator`, but this will be essential for testing https://github.com/oxidecomputer/omicron/issues/5552.

Reply:
What is the extra functionality?

Reply:
This lets you look at which resources were considered in-use at the time the iterator was constructed.